boot/zephyr/Kconfig

```diff
@@ -382,12 +382,16 @@ config BOOT_ENCRYPTION_KEY_FILE
 
 config BOOT_MAX_IMG_SECTORS
 	int "Maximum number of sectors per image slot"
+	default 512 if $(dt_nodelabel_reg_size_int,slot0_partition) > 1048576
+	default 256 if $(dt_nodelabel_reg_size_int,slot0_partition) > 524288
```

Comment:
This default value calculation is a good idea.
If you add it as a separate PR, it should be no complains.

Reply:
this really does not work well, for example a 2MiB stm32 device with 128 or 256KiB sectors here needs a value of 12 or 6 really, but would have a value of 512...

Reply:
> this really does not work well, for example a 2MiB stm32 device with 128 or 256KiB sectors here needs a value of 12 or 6 really, but would have a value of 512...

You are right. Sector size is missed :(

Reply:
> this really does not work well, for example a 2MiB stm32 device with 128 or 256KiB sectors here needs a value of 12 or 6 really, but would have a value of 512...

Does it work better than the current solution? Sector size is not exposed to the build or configuration system.

Reply:
> > this really does not work well, for example a 2MiB stm32 device with 128 or 256KiB sectors here needs a value of 12 or 6 really, but would have a value of 512...
> 
> Does it work better than the current solution? Sector size is not exposed to the build or configuration system.

Unfortunately without a sector size it has no big sense.

Reply:
So moving from:
* Works for 256kB image, 4kB sectors

to:

* Works for 256kB, 512kB, 1MB images, 4kB sectors

isn't valuable?

Reply:
> So moving from:
> 
> * Works for 256kB image, 4kB sectors
> 
> to:
> 
> * Works for 256kB, 512kB, 1MB images, 4kB sectors
> 
> isn't valuable?

Platforms have different sector sizes.

Reply:
I've created https://github.com/mcu-tools/mcuboot/pull/2019 though I've tested it on one device - nrf52840dk, it gets 118 which is about right. Haven't actually flashed it to test. It will also probably be slightly wrong if someone has a larger slot0 than slot1 using swap using move, but that would be 2 sectors it'd be out by